demos/src/Examples/Messaging/React/TextformattingMenu.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+
+const TextformattingMenu = ({ editor }) => {
+  if (!editor) {
+    return null
+  }
+
+  const addImage = () => {
+    const url = window.prompt('URL')
+
+    if (url) {
+      editor.chain().focus().setImage({ src: url }).run()
+    }
+  }
+
+  return (
+    <nav>
+      <button
+        onClick={() => editor.chain().focus().toggleBold().run()}
+        className={editor.isActive('bold') ? 'is-active' : ''}
+      >
+        bold
+      </button>
+      <button
+        onClick={() => editor.chain().focus().toggleItalic().run()}
+        className={editor.isActive('italic') ? 'is-active' : ''}
+      >
+        italic
+      </button>
+      <button
+        onClick={() => editor.chain().focus().toggleStrike().run()}
+        className={editor.isActive('strike') ? 'is-active' : ''}
+      >
+        strike
+      </button>
+      <div className="divider"></div>
+      <button onClick={addImage}>
+          add image
+      </button>
```

Comment:
I would not use the image extension at all in this demo. This only overcomplicates things here.